frontend/lib/src/components/widgets/DataFrame/columns/utils.ts

```diff
@@ -470,7 +470,13 @@ export function formatNumber(
   } else if (format === "duration[ns]") {
     return moment.duration(value / (1000 * 1000), "milliseconds").humanize()
   } else if (format.startsWith("period[")) {
-    return formatPeriodType(BigInt(value), format as any)
+    const match = format.match(/period\[(.*)]/)
+    if (match === null) {
+      return String(value)
+    }
+    const [, freq] = match
+
+    return formatPeriodFromFreq(value, freq as any)
```

Comment:
nit: instead of typing it as `any` here, could we also type it as `PeriodFrequency` which is what the function expects?

Reply:
Yep, that works. I changed it to `PeriodFrequency`